### DISCUSSION TITLE
Install Statistics Report

### DISCUSSION THREAD

**Discussion by DioMeo** [phax/peppol-reporting#6] [Q&A]
Hello, we have installed the version 6.3.0 of Oxalis with AS4 protocol.
We are using your SMP at version 6.0.7.
We should integrate the statistics report, as per the regulations, except that from the documentation found in this regard, that of Oxalis, would appear to be deprecated.
We wanted to know how we could integrate your peppol_reporting_library with our applications.
Is it possibile to do it?
Thanks.

**Comment by phax** [comment]
Hi @DioMeo, thanks for reaching out.
Peppol Reporting consists of 3 general phases:
1. Data collection
2. Data aggregation
3. Data sending

This library helps you primarily in step 2. For step 1 and 3 your AP solution needs to take care of it.

For exact details on HOW to integrate, I must admit I haven't done it before but it SHOULD be possible.

**Comment by RichardVanMaaren** [comment]
Hi @DioMeo,

Indeed the hardest part is gathering the data, In my example we store all transaction data in a DB. For outgoing, this is quite easy as you know what you are sending.

For Incoming this is more complicated and you could build ad plugin for oxalis which gathers the data, The example below is old but general flow can be used as example. https://github.com/RichardVanMaaren/oxalis-receipt-parser/tree/master

We are using it this way and the we fill the Helger peppol reporting library with data from the DB.

Hope this helps in some way.

Regards,

Richard

**Comment by DioMeo** [comment]
Hello, and thanks to reply me.
I created a MySQL db and configured in my .oxalis/oxalis.conf but it doesn't write nothing in db.
This my conf:
![immagine](https://github.com/phax/peppol-reporting/assets/146419702/e9b43bb0-91d0-488a-8833-100764881bc4)
I hope you can help me

**Reply by RichardVanMaaren** [reply]
Out of itself Oxalis does nothing. The plugin structure is only an example mentioned here.

For real Oxalis reporting questions you can refer to the gihub oxalis page.

Regards,

Richard

**Comment by DioMeo** [comment]
I took my conf from the example found in Oxalis, but it is not clear.
There were 2 issues where other people said that it doesn't work write on a db:
https://github.com/OxalisCommunity/oxalis/issues/347
https://github.com/OxalisCommunity/oxalis/issues/309

Thanks

**Reply by RichardVanMaaren** [reply]
Oxalis statistics is indeed legacy and very old.

In order to achieve what you want to store your data, you can create a plugin as mentioned with the example. Maybe an option to ask the same question on the Oxalis github page what other solutions there are ?

Regards,

Richard

**Reply by DioMeo** [reply]
I created an issue in Oxalis github but noone answered me.
